src/openrct2/object/RideObject.cpp

```diff
@@ -212,134 +226,33 @@ void RideObject::Load()
-                    b = vehicleEntry->base_num_frames * 40;
-                    image_index += b;
-                }
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_FLAT_TO_GENTLE_SLOPE_BANKED_TRANSITIONS)
-                {
-                    vehicleEntry->flat_to_gentle_bank_image_id = image_index;
-                    b = vehicleEntry->base_num_frames * (4 * numRotationFrames);
-                    image_index += b;
-                }
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_DIAGONAL_GENTLE_SLOPE_BANKED_TRANSITIONS)
-                {
-                    vehicleEntry->diagonal_to_gentle_slope_bank_image_id = image_index;
-                    b = vehicleEntry->base_num_frames * 16;
-                    image_index += b;
-                }
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_GENTLE_SLOPE_BANKED_TRANSITIONS)
-                {
-                    vehicleEntry->gentle_slope_to_bank_image_id = image_index;
-                    b = vehicleEntry->base_num_frames * 16;
-                    image_index += b;
-                }
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_GENTLE_SLOPE_BANKED_TURNS)
-                {
-                    vehicleEntry->gentle_slope_bank_turn_image_id = image_index;
-                    b = vehicleEntry->base_num_frames * (4 * numRotationFrames);
-                    image_index += b;
-                }
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_FLAT_TO_GENTLE_SLOPE_WHILE_BANKED_TRANSITIONS)
+                if (vehicleEntry->SpriteGroups[spriteGroup].Enabled())
                 {
-                    vehicleEntry->flat_bank_to_gentle_slope_image_id = image_index;
-                    b = vehicleEntry->base_num_frames * 16;
-                    image_index += b;
-                }
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_CORKSCREWS)
-                {
-                    vehicleEntry->corkscrew_image_id = image_index;
-                    b = vehicleEntry->base_num_frames * 80;
-                    image_index += b;
-                }
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_RESTRAINT_ANIMATION)
-                {
-                    vehicleEntry->restraint_image_id = image_index;
-                    b = vehicleEntry->base_num_frames * 12;
-                    image_index += b;
-                }
-
-                if (vehicleEntry->sprite_flags & VEHICLE_SPRITE_FLAG_CURVED_LIFT_HILL)
-                {
-                    // Same offset as corkscrew
-                    vehicleEntry->curved_lift_hill_image_id = image_index;
-                    b = vehicleEntry->base_num_frames * numRotationFrames;
-                    image_index += b;
+                    vehicleEntry->SpriteGroups[spriteGroup].imageId = imageIndex;
+                    spriteCountSwap = vehicleEntry->base_num_frames
```

Comment:
what does this even mean? Its just the sprite count afaik what does it mean "swap"?